src/Sulu/Bundle/AdminBundle/Resources/js/containers/FieldBlocks/FieldBlocks.js

```diff
@@ -13,6 +13,40 @@ const MISSING_BLOCK_ERROR_MESSAGE = 'The "block" field type needs at least one t
 const BLOCK_PREVIEW_TAG = 'sulu.block_preview';
 
 export default class FieldBlocks extends React.Component<FieldTypeProps<Array<BlockEntry>>> {
+    componentDidUpdate(prevProps: FieldTypeProps<Array<BlockEntry>>) {
+        const {defaultType, types, value: value, onChange} = this.props;
+        const {types: oldTypes} = prevProps;
+
+        if (!types || !oldTypes) {
+            throw new Error(MISSING_BLOCK_ERROR_MESSAGE);
+        }
+
+        const newValue = toJS(value);
+        let hasValueChanged = false;
+
+        if (value && types !== oldTypes) {
+            if (!defaultType) {
+                throw new Error(
+                    'It is impossible that a block has no defaultType. This should not happen and is likely a bug.'
+                );
+            }
+
+            // set block to default type if type does not longer exist
+            // this could happen for example in a template switch
+            newValue.forEach((block, i) => {
```

Comment:
This part is also different from what I've described in [another comment](https://github.com/sulu/sulu/pull/5099#discussion_r394340140).

Reply:
same as above I think if we change the value in componentDidUpdate we should call onChange only once and thats onChange is not hear and why I keep the forEach to set the boolean flag

Reply:
Ok, but as described above I would compare the old value with the new one, instead of introducing a new variable. Would also have the advantage that it would not call the `onChange` handler if a first block changes the value, and the second block reverts this changes for some reason.

I also think it is not necessary to use the `i` variable in here, because you should be able to just use `block.type = defaultType` below, without the need of accessing the array via the index.

Reply:
In case of performance I don't wanted to clone the whole object on every componentDidUpdate run, also when nothing necessary changed so I did go with only cloning the block object which where changed inside the `.map` function hope that solution match your criterias 🙂.